sway-fmt-v2/src/utils/where_clause.rs

```diff
@@ -0,0 +1,61 @@
+use crate::fmt::*;
+use std::fmt::Write;
+use sway_parse::{WhereBound, WhereClause};
+use sway_types::Spanned;
+
+impl Format for WhereClause {
+    fn format(
+        &self,
+        formatted_code: &mut FormattedCode,
+        formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        writeln!(
+            formatted_code,
+            "\n{}{}",
+            &formatter.shape.indent.to_string(formatter),
+            self.where_token.span().as_str(),
```

Comment:
Here we may want to check `ItemBraceStyle::SameLineWhere` here and place the `where` according to that.

Reply:
Right, this PR's scope is to get everything necessary for basic `fn` in. I think we can do this in a separate PR.

Reply:
Also, `SameLineWhere` is described as: 
```rust
/// Prefer the same line except where there is a where-clause, in which
/// case force the brace to be put on the next line.
```

so this shouldn't be handled inside `where_clause`

Reply:
It should actually be handled inside of `CurlyBrace::format()` I think

Reply:
I think we may benefit from having a function on `Shape` that returns a boolean value for a `WhereClause` case. I'll give it a test and update it after this PR since it's already grown to this size.